[docs] TV: include Expo CLI upgrade support (#46210)
expo/expo: docs/pages/guides/building-for-tv.mdx

### Modify dependencies for TV

In **package.json**, modify the `react-native` dependency to use the TV repo.

<Tabs>

<Tab label="SDK 56 and later">
```json package.json
{
  /* @hide ... */ /* @end */
  "dependencies": {
    /* @hide ... */ /* @end */
    "react-native": "npm:react-native-tvos@0.85-stable"
    /* @hide ... */ /* @end */
  }
}
```

The `react-native-tvos` version must match the Expo SDK you are using. For example, Expo SDK 56 uses React Native 0.85, so you should use `react-native-tvos@0.85-stable` (the latest 0.85 version) as shown above. See the [SDK compatibility table](/versions/latest/#each-expo-sdk-version-depends-on-a-react-native-version) for the correct version to use.

> In SDK 56 and later, the dependency on the TV repo will be correctly upgraded when [upgrading your project to a newer SDK version](/workflow/upgrading-expo-sdk-walkthrough/#upgrade-the-expo-sdk).

Reviewer: **[suggestion]** Passive future (`will be correctly upgraded`) where active present reads better (`expo-docs-style-guide.md#voice-and-tone`).

```suggestion
In SDK 56 and later, [upgrading your project to a newer SDK version](/workflow/upgrading-expo-sdk-walkthrough/#upgrade-the-expo-sdk) also upgrades the TV repo dependency.
```

}
```

The `react-native-tvos` version must match the Expo SDK you are using. For example, Expo SDK 55 uses React Native 0.83, so you should use `react-native-tvos@0.83-stable` (the latest 0.83 version) as shown above. See the [SDK compatibility table](/versions/latest/#each-expo-sdk-version-depends-on-a-react-native-version) for the correct version to use.

> **Warning** If upgrading an Expo TV project to SDK 55 or earlier, the `react-native-tvos` version **must be upgraded manually**. It will not be automatically updated by `npx expo install expo@latest --fix`.

</Tab>

</Tabs>

> **Warning** If you have more than one Expo project in a monorepo, and one of them is modified for TV, then all of them should be modified to use the React Native TV package as described here, even if some of the projects are not configured to target TV. This avoids possible conflicts between the project dependencies, while still supporting mobile development fully on all the projects.

> > {/* prettier-ignore */}

Reviewer 2: I'd check whether enabling `autolinkingModuleResolution` could be recommended here. In theory, since it includes `react-native` and that's remapped in the package manager, that should still work

Edit: oops, I mean, it's enabled in monorepos by default since SDK 55, so it's possible this isn't even a problem in SDK 55+

Author: Thank you!!!! I've already tested that upgrading from SDK 55 to 56 actually does the RNTV upgrade correctly when the expo.install.exclude is removed
